### PR TITLE
AG-12847 - more bundle size changes

### PR DESCRIPTION
**Pull request by peterjrreynolds**

### CODE REVIEW

File: packages/ag-grid-community/src/columns/columnGroupStateService.ts
Comment (StephenCooper) [inline packages/ag-grid-community/src/columns/columnGroupStateService.ts:1]: Needs deleting?

File: packages/ag-grid-community/src/ctrlsService.ts [hunk 1]
```diff
@@ -46,45 +46,16 @@ interface ReadyParams {
     rightHeader: HeaderRowContainerCtrl;
 }
 
+const NUM_CTRLS = 23;
```
Comment (StephenCooper) [inline packages/ag-grid-community/src/ctrlsService.ts:51]: What is 23?

File: packages/ag-grid-community/src/ctrlsService.ts [hunk 2]
```diff
@@ -113,9 +84,12 @@ export class CtrlsService extends BeanStub<'ready'> implements NamedBean {
         );
     }
     private updateReady(): void {
-        this.ready = Object.values(this.params).every((ctrl: BeanStub<any> | undefined) => {
-            return ctrl?.isAlive() ?? false;
-        });
+        const values = Object.values(this.params);
+        this.ready =
+            values.length === NUM_CTRLS &&
```
Comment (StephenCooper) [inline packages/ag-grid-community/src/ctrlsService.ts:94]: Debug logic?

File: packages/ag-grid-community/src/entities/positionUtils.ts [hunk 3]
```diff
@@ -63,75 +57,59 @@ export function _isSameRow(rowA: RowPosition | undefined, rowB: RowPosition | un
...
-    private pageBoundsService: PageBoundsService;
-    private rowRenderer: RowRenderer;
-
-    public wireBeans(beans: BeanCollection): void {
-        this.rowModel = beans.rowModel;
-        this.pinnedRowModel = beans.pinnedRowModel;
-        this.pageBoundsService = beans.pageBoundsService;
-        this.rowRenderer = beans.rowRenderer;
+export function _getFirstRow(beans: BeanCollection): RowPosition | null {
+    let rowIndex = 0;
```
Comment (StephenCooper) [inline packages/ag-grid-community/src/entities/positionUtils.ts:61]: Probably should deconstruct from beans for all of these methods to reduce duplication of access from beans.

File: packages/ag-grid-community/src/events.ts [hunk 4]
```diff
@@ -485,7 +485,7 @@ export interface RowDragEvent<TData = any, TContext = any, T extends AgEventType
     /** The `eventPath` persists the `event.composedPath()` result for access within AG Grid event handlers.  */
     eventPath?: EventTarget[];
     /** Direction of the drag, either `'up'`, `'down'` or `null` (if mouse is moving horizontally and not vertically). */
-    vDirection: string;
+    vDirection: string | null;
```
Comment (StephenCooper) [inline packages/ag-grid-community/src/events.ts]: Should this be typed accurately?

File: packages/ag-grid-enterprise/src/statusBar/providedPanels/aggregationComp.ts [hunk 5]
```diff
@@ -27,24 +24,23 @@ import {
 import type { AgNameValue } from './agNameValue';
 import { AgNameValueSelector } from './agNameValue';
 
-function _formatNumberTwoDecimalPlacesAndCommas(value: number, localeService: LocaleService): string {
+function _formatNumberTwoDecimalPlacesAndCommas(
+    value: number,
+    getLocaleTextFunc: () => (key: string, defaultValue: string, variableValues?: string[]) => string
```
Comment (StephenCooper) [inline packages/ag-grid-enterprise/src/statusBar/providedPanels/aggregationComp.ts]: Is it worth making a type for this as it is in a few places?

File: packages/ag-grid-community/src/events.ts (hunk 4 above)
Comment (peterjrreynolds) [reply packages/ag-grid-community/src/events.ts]: This fixes an existing bug - see AG-13170

File: packages/ag-grid-community/src/columns/columnGroupStateService.ts
Comment (peterjrreynolds) [reply packages/ag-grid-community/src/columns/columnGroupStateService.ts:1]: It has been deleted

File: packages/ag-grid-community/src/ctrlsService.ts (hunk 1 above)
Comment (peterjrreynolds) [reply packages/ag-grid-community/src/ctrlsService.ts:51]: The number of controls. Saves us having to pre-populate the params object. Have added some comments

File: packages/ag-grid-community/src/ctrlsService.ts (hunk 2 above)
Comment (peterjrreynolds) [reply packages/ag-grid-community/src/ctrlsService.ts:94]: See other comment

File: packages/ag-grid-enterprise/src/statusBar/providedPanels/aggregationComp.ts (hunk 5 above)
Comment (peterjrreynolds) [reply packages/ag-grid-enterprise/src/statusBar/providedPanels/aggregationComp.ts]: Done

File: packages/ag-grid-community/src/entities/positionUtils.ts (hunk 3 above)
Comment (peterjrreynolds) [reply packages/ag-grid-community/src/entities/positionUtils.ts:61]: Done

**Pull request merged**